build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.jpms-modules.gradle.kts

```diff
@@ -14,284 +14,209 @@
-    withModule<RemoveAnnotationLibrariesMetadataRule>("com.google.dagger:dagger-compiler")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("com.google.dagger:dagger-producers")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("com.google.dagger:dagger-spi")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("com.google.guava:guava")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("com.google.protobuf:protobuf-java-util")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("io.helidon.grpc:io.grpc")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("org.apache.tuweni:tuweni-bytes")
-    withModule<RemoveAnnotationLibrariesMetadataRule>("org.apache.tuweni:tuweni-units")
-
-    withModule<IoNettyNativeEpollMetadataRule>("io.netty:netty-transport-native-epoll")
-
-    withModule<IoPrometheusSimpleclientMetadataRule>("io.prometheus:simpleclient")
-
-    withModule<RemoveKotlinStdlibCommonMetadataRule>("org.jetbrains.kotlin:kotlin-stdlib")
+    withModule<AddFeaturesMetadataRule>("io.netty:netty-transport-native-epoll") {
+        params(listOf("linux-x86_64", "linux-aarch_64"))
+    }
 
-    withModule<RemoveHamcrestCoreMetadataRule>("junit:junit")
-}
+    // The following 'io.grpc' libraries are replaced with a singe dependency to
+    // 'io.helidon.grpc:io.grpc', which is a re-packaged Modular Jar of all the 'grpc' libraries.
+    val grpcComponents = listOf("io.grpc:grpc-api", "io.grpc:grpc-context", "io.grpc:grpc-core")
+    val grpcModule = listOf("io.helidon.grpc:io.grpc")
 
-extraJavaModuleInfo {
-    failOnAutomaticModules.set(true) // Only allow Jars with 'module-info' on all module paths
+    // These compile time annotation libraries are not of interest in our setup and are thus removed
+    // from the dependencies of all components that bring them in.
+    val annotationLibraries =
+        listOf(
+            "com.google.android:annotations",
+            "com.google.code.findbugs:annotations",
+            "com.google.code.findbugs:jsr305",
+            "com.google.errorprone:error_prone_annotations",
+            "com.google.guava:listenablefuture",
+            "com.google.j2objc:j2objc-annotations",
+            "org.checkerframework:checker-compat-qual",
+            "org.checkerframework:checker-qual",
+            "org.codehaus.mojo:animal-sniffer-annotations"
+        )
 
-    module("io.grpc:grpc-netty", "grpc.netty") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
-        requires("java.logging")
-    }
-    module("io.grpc:grpc-stub", "grpc.stub") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
-        requires("java.logging")
-    }
-    module("io.grpc:grpc-testing", "grpc.testing") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
+    withModule<RemoveDependenciesMetadataRule>("io.grpc:grpc-netty") {
+        params(grpcComponents + annotationLibraries)
```

Comment:
Use standard rule implementations here, so it is easier to read. The separate rule Koltin classes are removed and all information about what is modified in the metadata is found directly here in this file.